decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -72,26 +41,31 @@ class RenamingScheme(ABC):
     """Base class for different Renaming schemes."""
 
     def __init__(self, task: DecompilerTask) -> None:
-        """Collets all needed variables for renaming + filters already renamed + function arguments out"""
-        collector = VariableCollector(task._ast.condition_map)
+        """Collets all needed variables for renaming + filters which should not be renamed"""
+        collector = VariableCollector()
         collector.visit_ast(task._ast)
+        self._ast = task._ast
         self._params: List[Variable] = task._function_parameters
-        self._loop_vars : List[Variable] = collector.get_loop_variables()
-        self._variables: List[Variable] = list(filter(self._filter_variables, collector.get_variables()))
+        self._variables: Set[Variable] = set(filter(self._filter_variables, collector.get_variables()))
         
 
     def _filter_variables(self, item: Variable) -> bool:
         """Return False if variable is either a:
             - parameter
-            - renamed loop variable
             - GlobalVariable
         """
-        return not item in self._params and not (item in self._loop_vars and item.name.find("var_") == -1) and not isinstance(item, GlobalVariable)
+        return not item in self._params and not isinstance(item, GlobalVariable)
+
+
+    def renameVariables(self):
+        """Rename all collected variables with a naming scheme."""
+        for var in self._variables:
+            self._ast.replace_variable_in_subtree(self._ast.root, var, Variable(self.getVariableName(var), var.type))
```

Comment:
What is with all the other information of a variable:
```python
        ssa_label: Optional[int] = None,
        is_aliased: bool = False,
        ssa_name: Optional[Variable] = None,
        tags: Optional[Tuple[Tag, ...]] = None,
```

Reply:
- `ssa_label/ssa_name` are always None because `out-of-ssa` removed them
- `is_aliased` should be added for correctness (although the stages after name generation don't use them anyway)
- `tags` are only used by tagging which is removed in pre-processing

Reply:
`ssa_label` is `None`, but `ssa_name` is not None. We probably do not need it after the restrucutring, but not sure.